include/astshim/ChebyMap.h

```diff
@@ -0,0 +1,321 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef ASTSHIM_CHEBYMAP_H
+#define ASTSHIM_CHEBYMAP_H
+
+#include <algorithm>
+#include <memory>
+#include <vector>
+
+#include "astshim/base.h"
+#include "astshim/Mapping.h"
+
+namespace ast {
+
+class ChebyDomain {
+public:
+    /**
+    Construct a ChebyDomain
+
+    @param[in] lbnd  Lower bounds of domain (one element per axis)
+    @param[in] ubnd  Upper bounds of domain (one element per axis)
+    */
+    ChebyDomain(std::vector<double> const &lbnd, std::vector<double> const &ubnd) : lbnd(lbnd), ubnd(ubnd) {}
+    std::vector<double> lbnd;  ///< lower bound of domain (one element per axis)
+    std::vector<double> ubnd;  ///< upper bound of domain (one element per axis)
```

Comment:
If this is just a passive container for these vectors, it should probably be declared as a struct, not a class.